File: Assembler/Python/Ainstruction.py
f = open("symboltable.asm","r")
f2 = open("A_instruction.hack","w")
f3 = open("whitespace.asm","r")
lines = f.readlines()
A=[]
B=[]
for i in lines:
    i=i.replace("\n","")
    k=i.split(",")
    A.append(k[0])
    B.append(k[1])
lines2 = f3.readlines()
for i in lines2:
    i=i.replace("\n","")
    i = i.replace(")","")
    p = i[1:]
    if i[0]=='@' and i[1].isnumeric():
        f2.writelines(str(bin(int(i[1:]))[2:]).zfill(16)+'\n')
    for j in range(len(A)):
        if p==A[j]:
            f2.writelines((bin(int(B[j]))[2:]).zfill(16)+'\n')

File: Assembler/Java/Ainstruction.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;
import java.io.FileWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
public class Ainstruction {
    public static void main(String[] args) throws IOException {
        try{
            File read=new File("nowhitespaces.asm");            
            Scanner in=new Scanner(read);
            List<String> variables= new ArrayList<String>();
            FileWriter writer=new FileWriter("Ainstruction.hack"); 
            while (in.hasNextLine()) {                                                   
                String data = in.nextLine();
                if(data.startsWith("@"))
                {
                    int flag=0;
                    File symbol=new File("symboltable.asm");
                    Scanner scanner=new Scanner(symbol);
                    while(scanner.hasNextLine())
                    {
                        String symbo = scanner.nextLine();
                        data=data.replaceAll(" ", "");
                        symbo=symbo.replaceAll(" ", "");
                        if(data.substring(1).equals(symbo.substring(0,symbo.indexOf(":"))))
                        {
                            int c=symbo.indexOf(":")+1;
                            String val=(symbo.substring(c));
                            variables.add(val);
                            flag++;
                        }
                    }
                    scanner.close();
                    if(flag==0)
                    {
                        variables.add(data.substring(1));
                    }
                }
            }
            for(String variable : variables)                                              // iliterating values in variable (for each loop)
            {
                try {
                    int number = Integer.parseInt(variable);
                    String binary = Integer.toBinaryString(number);
                    // Pad with leading zeros to ensure 16 bits
                    binary = String.format("%16s", binary).replace(' ', '0');
                    writer.write(binary+'\n');
                } catch (NumberFormatException e) {
                    System.out.println(variable);                
                }
            }
            writer.close();
            in.close();
        } catch(FileNotFoundException e)
        {                                            
            System.out.println("An error occured.");
            e.printStackTrace();
        }
    }
}